### ISSUE TITLE
Misinterpretation in DL3010

### ISSUE THREAD

**Issue by razamat** [hadolint/hadolint#709]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
In my Dockerfile, I want to copy *.tar.gz archive to a docker image. This archive will be later used by a *.sh script.
There is no reason for me to unpack the archive, I need it in packed form.
```COPY project/build/data/data.tar.gz /data/```

In my understanding, DL3010 should appear if hadolint would detect 
```COPY <archive_name>``` followed by a ```RUN tar -xf <same_archive_name>``` or something similar.

### Actual behavior
Using command ```COPY``` with a *.tar.gz file cause a complain about DL3010. 
Using command ```ADD``` as hadolint suggests, unpacks the archive.

There are a few possible workarounds like ```COPY /whole_directory/``` so hadolint doesn't notice the archive
or using ```ADD```  and "double packing" the file or simply ordering hadolint to ignore DL3010.

### Steps to reproduce the behavior
Use command COPY with a *.tar.gz file.

```bash
[2021-09-10T03:07:22.183Z] Dockerfile:53 DL3010 info: Use ADD for extracting archives into an image
```

**Dockerfile** (if relevant)

```Dockerfile
FROM [redacted]

RUN [redacted] \
 && mkdir -p /data

COPY project/build/script/ /bin/
COPY project/build/data/data.tar.gz /data/

CMD ["/bin/start.sh"]
```

**Comment by m-ildefons** [comment]
Hi @razamat,
if you have good reason to ignore a particular rule, like in your case, you can also annotate that instruction to ignore that rule. E.g.:
```
[...]
# hadolint ignore=DL3010
COPY project/build/data/data.tar.gz /data/
[...]
```
Then Hadolint will omit that rule on that line. See [here](https://github.com/hadolint/hadolint#inline-ignores) for more information.
You can also disable rules entirely if you are sure that they will never be relevant to you.

**Comment by razamat** [comment]
Thanks @m-ildefons,
I actually didn't know about "inline ignores", these might come handy in the future.

It's true that there are workarounds, "ignore" one and couple of others.
Nevertheless, I just found DL3010 to be very restrictive as copying an archive and not unpacking it can be a valid action.

**Comment by m-ildefons** [comment]
Hi @razamat ,
I've revisited the rule and constructed logic to detect a `COPY something.tar.gz` followed by a `RUN tar -xf something.tar.gz` just like you proposed and I think this should solve the issue. Lets see if it will be merged.

**Comment by razamat** [comment]
Great!
Thank you very much @m-ildefons.
